server/src-lib/Hasura/GraphQL/Execute.hs

```diff
@@ -109,40 +104,17 @@ getExecPlanPartial userInfo sc enableAL req
   when enableAL checkQueryInAllowlist
   (gCtx, _) <- flip runStateT sc $ getGCtx role gCtxRoleMap
   queryParts <- flip runReaderT gCtx $ VQ.getQueryParts req
-  let remoteSchemas = scRemoteSchemas sc
   rootSelSet <- runReaderT (VQ.validateGQ queryParts) gCtx
-  runReaderT (generatePlan rootSelSet) (gCtx, remoteSchemas)
+  return $ uncurry (GQExecPlanPartial gCtx) $ generatePlan rootSelSet
```

Comment:
maybe now we can call this function something other than `generatePlan` like say `uncurryFields` or something